front-end/src/components/Products.jsx

```diff
@@ -1,11 +1,91 @@
-import React from 'react';
+import React, { useEffect, useState, useContext } from 'react';
+import axios from 'axios';
+import './style/Products.css';
+import { BeerContext } from '../context/context';
 import MenuTop from './MenuTop';
+import { Redirect, Link } from 'react-router-dom';
+import CardProducts from './CardProducts';
+
+const instance = axios.create({ baseURL: 'http://localhost:3001' });
+
+const addTobascket = (price, name, setCart) => {
+  const item = { name, price };
+  setCart(currentState => [...currentState, item]);
+  const cart = localStorage.getItem('cart');
+  if (cart) {
+    localStorage.setItem('cart', JSON.stringify([
+      ...JSON.parse(cart), {
+        name, price,
+      }]));
+  } else {
+    localStorage.setItem('cart', JSON.stringify([{
+      name, price,
+    }]));
+  }
+}
+
+const removeToBascket = (cart, name, setCart) => {
+  if (cart.length === 0) return;
+  const remove = cart.filter(e => e.name !== name);
+  localStorage.setItem('cart', JSON.stringify(remove))
+  setCart(remove);
+}
```

Comment:
Willian, aqui quando é apertado o botão de diminuir ele remove o carrinho todo por que o filter remove tudo que ele encontra e que seja igual, até procurei algo que desse pra remover apenas um, mas só consegui via reduce

```suggestion
const removeToBascket = (cart, name, setCart) => {
  if (cart.length === 0) return;
  const remove = cart.filter(e => e.name !== name);
  localStorage.setItem('cart', JSON.stringify(remove))
  setCart(remove);
}const removeToBascket = (cart, name, setCart) => {
  if (cart.length === 0) return;
  const remove = cart.reduce((acc, e) => {
    if (e.name === name) {
      if (acc.counter > 0) {
        acc.arr = [...acc.arr, e];
        return acc;
      }
      acc.counter = 1;
      return acc;
    }
    acc.arr = [...acc.arr, e];
    return acc;
  }, { arr: [], counter: 0 });
  localStorage.setItem('cart', JSON.stringify(remove.arr));
  setCart(remove.arr);
};
```